File: python/CompoundFraction.py
#!/usr/bin/python
from maths import *

class CompoundFraction:
  def __init__(self, numerator, denominator, whole):
    self.numerator = numerator
    self.denominator = denominator
    self.whole = whole

def compoundFractionToFraction(cf):
  result = Fraction(cf.whole * cf.denominator + cf.numerator, cf.denominator)
  return result

def fractionToCompoundFraction(f):
  whole = math.floor(f.numerator / f.denominator)
  numerator = f.numerator % f.denominator
  denominator = f.denominator
  result = CompoundFraction(whole, numerator, denominator)
  return result

File: c/math/mathFramework/CompoundFraction.c
#include <stdio.h>
#include <math.h>

#include "CompoundFraction.h"

static struct CompoundFraction new (int numerator, int denominator, int whole) {
	return (struct CompoundFraction) {
		.numerator = numerator, .denominator = denominator, .whole = whole
	};
}

const struct CompoundFractionClass CompoundFraction = {
	.new = &new
};

fract compoundFractionToFraction(cfract a) {
	return Fraction.new(a.whole * a.denominator + a.numerator, a.denominator);
}

cfract fractionToCompoundFraction(fract a) {
	return CompoundFraction.new(fmod(a.numerator, a.denominator), a.denominator, floor(a.numerator / a.denominator));
}
